### ISSUE TITLE
CUSTOM API ENDPOINT URL

### ISSUE THREAD

**Issue by Only1337** [jbexta/AgentPilot#9]
Is there a feature to add a custom endpoint for the API? If not, are you considering adding it?

**Comment by jbexta** [comment]
Yeah next version will have litellm so will be possible, will be released in a week or 2

**Comment by Only1337** [comment]
Got it! Thank you, @jbexta, for the quick response! I'm excited to hear that the next version will include the feature I requested. Looking forward to the release in a week or two.

**Comment by ishaan-jaff** [comment]
@jbexta @Only1337 what do you need from litellm ? happy to bumo urgency on this

**Comment by Only1337** [comment]
Hi @ishaan-jaff, thank you for your willingness to help. I’m currently hosting my own language model using the reverse proxy method and I need to add my custom endpoint URL through the settings. @jbexta has kindly informed that this feature will be available in the next version, which is expected to be released in a week or two. I’m looking forward to this update. Thanks again for your offer to expedite the process, it’s greatly appreciated!

**Comment by ishaan-jaff** [comment]
@Only1337 is your LLM OpenAI compatible @jbexta ?

**Comment by ishaan-jaff** [comment]
we support this: https://docs.litellm.ai/docs/providers/custom_openai_proxy

**Comment by jbexta** [comment]
@ishaan-jaff Hi ishaan I should be able to plug right into it, I think I've figured out the gui for the seettings for best control. Is there a csv or something of all the models, no worries if not. @Only1337 I'm not very clued up on local models or custom endpoints, I was assuming that link is what you meant ^

**Comment by ishaan-jaff** [comment]
@jbexta https://docs.litellm.ai/docs/providers

**Comment by Only1337** [comment]
Hey @jbexta, 
This is exactly what I was looking for. However, I believe there should also be support for keys in custom endpoints. If this feature could be added, I would be extremely grateful. Thank you for your hard work and dedication to this project.

**Comment by Only1337** [comment]
> @Only1337 is your LLM OpenAI compatible @jbexta ?

@ishaan-jaff Yes, my LLM is compatible with OpenAI.

**Comment by jbexta** [comment]
@Only1337 Hi yes no worries is it just the api_base and custom_provider fields you need? Thanks for your appreciation for the project 🙏

**Comment by Only1337** [comment]
Yes, I need api_base and custom_provider fields. Thank you for taking care of me.

**Comment by jbexta** [comment]
@Only1337 This might work now in version 0.1.0, but I've only tested openai and perplexity. There's a silent error with the linux build for some machines, will bring out patch versions soon

**Comment by Only1337** [comment]
It worked on my machine, but for some reason I couldn't add a custom model. The information I entered was not saved, most likely due to my incompetence. Also, I would like to express my gratitude for the release of the new version and your interest.

**Comment by jbexta** [comment]
> It worked on my machine, but for some reason I couldn't add a custom model. The information I entered was not saved, most likely due to my incompetence. Also, I would like to express my gratitude for the release of the new version and your interest.

@Only1337 hmm I will check, did you have an API selected when adding a new one?  No worries 👍
